Crash when hovering on methods with markdown definition lists

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When hovering over some python methods, such as `fixture` on `@pytest.fixture` the markdown changes from https://github.com/zed-industries/zed/pull/17418 trigger a crash due to definition lists being unimplemented.

I'm not entirely sure how to create a markdown that specifically trigger this, but it can be reproduced via:

```
python -m venv ./.venv
```
Setup pyrightconfig.json
```
{
  "venvPath": "<path_to_folder>",
  "venv": ".venv"
}
```
Create a conftest.py
```
import pytest

@pytest.fixture
def test()
...
```

Hover on `.fixture`, triggers crash.

### Environment

Zed: v0.153.0 (Zed Nightly c92a49e7aea41d125f7cddcdd00d2fa98184e935)
OS: macOS 14.6.1
Memory: 36 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->
2024-09-09T10:27:41.200602-07:00 [ERROR] {
  "thread": "<unnamed>",
  "payload": "not implemented: definition lists are not yet supported",
  "location_data": {
    "file": "crates/markdown/src/parser.rs",
    "line": 323
  },
  "backtrace": [
    "<backtrace::capture::Backtrace>::create",
    "<backtrace::capture::Backtrace>::new",
    "zed::reliability::init_panic_hook::{closure#0}",
    "std::panicking::rust_panic_with_hook",
    "std::panicking::begin_panic_handler::{{closure}}",
    "std::sys_common::backtrace::__rust_end_short_backtrace",
    "_rust_begin_unwind",
    "core::panicking::panic_fmt",
    "<markdown::parser::MarkdownTag as core::convert::From<pulldown_cmark::Tag>>::from",
    "markdown::parser::parse_markdown",
    "<markdown::Markdown>::parse::{closure#0}",
    "<async_task::raw::RawTask<core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output = core::result::Result<markdown::ParsedMarkdown, anyhow::Error>> + core::marker::Send>>, core::result::Result<markdown::ParsedMarkdown, anyhow::Error>, <gpui::executor::BackgroundExecutor>::spawn_internal<core::result::Result<markdown::ParsedMarkdown, anyhow::Error>>::{closure#0}, ()>>::run",
    "_pthread_mach_thread_np"
  ],
  "app_version": "0.153.0",
  "release_channel": "Zed Nightly",
  "os_name": "macOS",
  "os_version": "14.6.1",
  "architecture": "aarch64",
  "panicked_on": 1725902861200,
  "installation_id": "80fb8d24-f9cb-440b-a64a-8633eaf03660",
  "session_id": "cbdeb994-7bc4-409d-95c0-8d3d699eaa36"
}
<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

--- a/crates/markdown/src/parser.rs
+++ b/crates/markdown/src/parser.rs
@@ -232,6 +232,10 @@ pub enum MarkdownTag {
 
     /// A metadata block.
     MetadataBlock(MetadataBlockKind),
+
+    DefinitionList,
+    DefinitionListTitle,
+    DefinitionListDefinition,
 }
 
 #[derive(Clone, Debug, PartialEq)]
@@ -317,11 +321,9 @@ impl From<pulldown_cmark::Tag<'_>> for MarkdownTag {
             },
             pulldown_cmark::Tag::HtmlBlock => MarkdownTag::HtmlBlock,
             pulldown_cmark::Tag::MetadataBlock(kind) => MarkdownTag::MetadataBlock(kind),
-            pulldown_cmark::Tag::DefinitionList
-            | pulldown_cmark::Tag::DefinitionListTitle
-            | pulldown_cmark::Tag::DefinitionListDefinition => {
-                unimplemented!("definition lists are not yet supported")
-            }
+            pulldown_cmark::Tag::DefinitionList => MarkdownTag::DefinitionList,
+            pulldown_cmark::Tag::DefinitionListTitle => MarkdownTag::DefinitionListTitle,
+            pulldown_cmark::Tag::DefinitionListDefinition => MarkdownTag::DefinitionListDefinition,
         }
     }
 }

